Content.Server/GameObjects/Components/Mobs/Speech/SpanishAccentComponent.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.CodeAnalysis.Operations;
+using Microsoft.EntityFrameworkCore.Internal;
+using Robust.Shared.GameObjects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Content.Server.GameObjects.Components.Mobs.Speech
+{
+    [RegisterComponent]
+    public class SpanishAccentComponent : Component, IAccentComponent
+    {
+        public override string Name => "SpanishAccent";
+
+        public string Accentuate(string message)
+        {
+            // Insert E before every S
+            message = InsertS(message);
+            // If a sentence ends with ?, insert a reverse ? at the beginning of the sentence
+            message = ReplaceQuestionMark(message);
+            return message;
+        }
+
+        private static string InsertS(string message)
+        {
+            // Replace every new Word that starts with s/S
+            var msg = message.Replace(" s", " es").Replace(" S", "Es");
+
+            // Still need to check if the beginning of the message starts
+            if (msg.StartsWith("s"))
+            {
+                return msg.Remove(0, 1).Insert(0, "es");
+            }
+            else if (msg.StartsWith("S"))
+            {
+                return msg.Remove(0, 1).Insert(0, "Es");
+            }
+
+            return msg;
+        }
+
+        private static readonly Regex SentenceRegex = new Regex(@"(?<=[\.!\?])");
```

Comment:
Move this to a more generic place, like AccentManager or AccentHelpers or something of the sort as it will probably be useful for more accents.